ComAmazonawsDynamodb/runtimes/python/src/com_amazonaws_dynamodb/internaldafny/extern/Com_Amazonaws_Dynamodb.py

```diff
@@ -0,0 +1,57 @@
+import boto3
+import _dafny
+
+from botocore.config import Config
+from boto3.session import Session
+
+from standard_library.internaldafny.generated.Wrappers import Option_Some
+from com_amazonaws_dynamodb.smithygenerated.com_amazonaws_dynamodb.shim import DynamoDBClientShim
+from com_amazonaws_dynamodb.internaldafny.generated.Com_Amazonaws_Dynamodb import *
+import com_amazonaws_dynamodb.internaldafny.generated.Com_Amazonaws_Dynamodb
+
+
+# Persist this across calls; this doesn't change
+available_aws_regions: list[str] = None
+
+def get_available_aws_regions():
+    global available_aws_regions
+    if available_aws_regions is not None:
+        return available_aws_regions
+    available_aws_regions = Session().get_available_regions("dynamodb")
+    return available_aws_regions
+
+class default__(com_amazonaws_dynamodb.internaldafny.generated.Com_Amazonaws_Dynamodb.default__):
+    @staticmethod
+    def DynamoDBClient(boto_client = None, region = None):
+        if boto_client is None:
+            if region is not None and region in get_available_aws_regions():
+                boto_config = Config(
+                    region_name=region
+                )
+                boto_client = boto3.client("dynamodb", config=boto_config)
+            else:
+                boto_client = boto3.client("dynamodb")
+                region = Session().region_name
```

Comment:
Question: Why not `boto_client.meta.region_name`?

Reply:
Good catch, will update